app/src/main/java/org/groundplatform/android/ui/map/gms/features/LineStringRenderer.kt

```diff
@@ -108,4 +107,20 @@ constructor(
     val strokeScale = if (selected) 2f else 1f
     return defaultStrokeWidth * strokeScale
   }
+
+  private fun processCoordinates(geometry: LineString): List<LatLng> {
+    val coordinates = geometry.coordinates.toLatLngList()
+    // If all points are identical, we add a tiny offset to the last point to ensure the
+    // polyline has non-zero length.
+    if (coordinates.size > 1 && coordinates.all { it == coordinates[0] }) {
+      return coordinates.dropLast(1) + coordinates.last().addOffset()
+    }
+    return coordinates
+  }
```

Comment:
Just to confirm, this is only possible when the size is 2 and both are same, right? In that case, let's update the condition to reflect the same.

Also, let's update the method name to better reflect the purpose (e.g.  `convertMaybeAddOffsetPoint`)